app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingPagingAdapter.kt

```diff
@@ -0,0 +1,134 @@
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.adapter.StatusBaseViewHolder
+import com.keylesspalace.tusky.adapter.TagViewHolder
+import com.keylesspalace.tusky.interfaces.LinkListener
+import com.keylesspalace.tusky.util.StatusDisplayOptions
+import com.keylesspalace.tusky.viewdata.TrendingViewData
+
+class TrendingPagingAdapter(
+    private var statusDisplayOptions: StatusDisplayOptions,
+    private val trendingListener: LinkListener,
+) : ListAdapter<TrendingViewData, RecyclerView.ViewHolder>(TrendingDifferCallback) {
+
+    var mediaPreviewEnabled: Boolean
+        get() = statusDisplayOptions.mediaPreviewEnabled
+        set(mediaPreviewEnabled) {
+            statusDisplayOptions = statusDisplayOptions.copy(
+                mediaPreviewEnabled = mediaPreviewEnabled
+            )
+        }
+
+    init {
+        stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
+    }
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            VIEW_TYPE_TAG -> {
+                val view = LayoutInflater.from(viewGroup.context)
+                    .inflate(R.layout.item_trending, viewGroup, false)
+                TagViewHolder(view)
+            }
+
+            else -> {
+                val view = LayoutInflater.from(viewGroup.context)
+                    .inflate(R.layout.item_trending, viewGroup, false)
+                TagViewHolder(view)
+            }
+        }
+    }
+
+    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
+        bindViewHolder(viewHolder, position, null)
+    }
+
+    override fun onBindViewHolder(
+        viewHolder: RecyclerView.ViewHolder,
+        position: Int,
+        payloads: List<*>
+    ) {
+        bindViewHolder(viewHolder, position, payloads)
+    }
+
+    private fun bindViewHolder(
+        viewHolder: RecyclerView.ViewHolder,
+        position: Int,
+        payloads: List<*>?
+    ) {
+        val trending = getItem(position)
+        if (trending is TrendingViewData.Tag) {
+            this.currentList
+
+            val maxTrendingValue = currentList
+                .flatMap { trendingViewData ->
+                    trendingViewData.asTagOrNull()?.tag?.history ?: emptyList()
+                }
+                .mapNotNull { it.accounts.toIntOrNull() }
+                .maxOrNull() ?: 1
+
+            val holder = viewHolder as TagViewHolder
+            holder.setup(trending, maxTrendingValue, trendingListener)
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return if (getItem(position) is TrendingViewData.Tag) {
+            VIEW_TYPE_TAG
+        } else {
+            VIEW_TYPE_TAG
+        }
+    }
+
```

Comment:
`override fun getItemViewType(position: Int): Int = VIEW_TYPE_TAG` ?

Reply:
The same as when above. I kept this here as a reminder that other Trending types exist. However, if that isn't clean enough I can remove it. Just let me know.

Reply:
Now there is the header type, so there is a real usage for it.